transport/src/main/java/io/netty/channel/AbstractCoalescingBufferQueue.java

```diff
@@ -291,6 +308,17 @@ private void incrementReadableBytes(int increment) {
         if (nextReadableBytes < readableBytes) {
             throw new IllegalStateException("buffer queue length overflow: " + readableBytes + " + " + increment);
         }
+        if (tracker != null) {
+            tracker.incrementPendingOutboundBytes(increment);
+        }
         readableBytes = nextReadableBytes;
```

Comment:
update the state of this class before calling out to `tracker` ... tracker may call into user code which may result in re-entry.